ark-blockstore-example/noop-plugin.go

```diff
@@ -0,0 +1,168 @@
+	return &NoOpBlockStore{logger: plugin.NewLogger()}
+}
+
+func (p *NoOpBlockStore) log(msg string, args ...interface{}) {
+	p.logger.Infof(msg, args...)
+}
+
+func (p *NoOpBlockStore) err(msg string, args ...interface{}) {
+	fmt.Printf("ERROR "+msg, args...)
+}
+
+// Init the plugin
+func (p *NoOpBlockStore) Init(config map[string]string) error {
+	p.log("Init called", config)
+	p.config = config
+
+	// Make this plugin re'initializable
+	p.volumes = make(map[string]Volume)
+	p.snapshots = make(map[string]Snapshot)
+	return nil
+}
+
+// CreateVolumeFromSnapshot Create a volume form given snapshot
+func (p *NoOpBlockStore) CreateVolumeFromSnapshot(snapshotID, volumeType, volumeAZ string, iops *int64) (string, error) {
+
+	p.log("CreateVolumeFromSnapshot called", snapshotID, volumeType, volumeAZ, *iops)
+	var volumeID string
+	for {
+		volumeID := snapshotID + ".vol." + strconv.FormatUint(rand.Uint64(), 10)
+		if _, ok := p.volumes[volumeID]; ok {
+			// Duplicate ? Retry
+			continue
+		}
+		break
+	}
+
+	p.volumes[volumeID] = Volume{
+		volType: volumeType,
+		az:      volumeAZ,
+		// TODO: See https://github.com/heptio/ark/issues/814
+		iops: *iops,
+	}
+	*iops = 100
+	return volumeID, nil
+}
+
+// GetVolumeInfo Get information about the volume
+func (p *NoOpBlockStore) GetVolumeInfo(volumeID, volumeAZ string) (string, *int64, error) {
+	p.log("GetVolumeInfo called", volumeID, volumeAZ)
+	if val, ok := p.volumes[volumeID]; ok {
+		// TODO: See https://github.com/heptio/ark/issues/814
+		iops := val.iops
+		return val.volType, &iops, nil
+	}
+	return "", nil, errors.New("Volume " + volumeID + " not found")
+}
+
+// IsVolumeReady Check if the volume is ready
+func (p *NoOpBlockStore) IsVolumeReady(volumeID, volumeAZ string) (ready bool, err error) {
+	p.log("IsVolumeReady called", volumeID, volumeAZ)
+	return true, nil
+}
+
+// CreateSnapshot Create a snapshot
+func (p *NoOpBlockStore) CreateSnapshot(volumeID, volumeAZ string, tags map[string]string) (string, error) {
+	p.log("CreateSnapshot called", volumeID, volumeAZ, tags)
+	var snapshotID string
+	for {
+		snapshotID = volumeID + ".snap." + strconv.FormatUint(rand.Uint64(), 10)
+		p.log("CreateSnapshot trying to create snapshot", snapshotID)
+		if _, ok := p.snapshots[snapshotID]; ok {
+			// Duplicate ? Retry
+			continue
+		}
+		break
+	}
+
+	// Remember the "original" volume, only required for the first
+	// time.
+	if _, exists := p.volumes[volumeID]; !exists {
+		p.volumes[volumeID] = Volume{
+			volType: "orignalVolumeType",
+			az:      volumeAZ,
+			iops: 100,
+		}
+	}
+
+	// Remember the snapshot
+	p.snapshots[snapshotID] = Snapshot{volID: volumeID,
+			az:   volumeAZ,
+			tags: tags}
+
+
+	p.log("CreateSnapshot returning", snapshotID)
+	return snapshotID, nil
+}
+
+// DeleteSnapshot Delete a snapshot
+func (p *NoOpBlockStore) DeleteSnapshot(snapshotID string) error {
+	p.log("DeleteSnapshot called", snapshotID)
+	if _, ok := p.snapshots[snapshotID]; ok {
```

Comment:
I would simplify this function to just `delete(p.snapshots, snapshotID)` and return a nil error. This is semantically consistent with our other implementations.

Reply:
Correct, and correct.  A better specification of the expectations for each function are in order.